homeassistant/components/playstation_network/__init__.py

```diff
@@ -0,0 +1,52 @@
+"""The Playstation Network integration."""
+
+from __future__ import annotations
+
+from psnawp_api.core.psnawp_exceptions import PSNAWPAuthenticationError
+from psnawp_api.psnawp import PSNAWP
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .coordinator import PlaystationNetworkCoordinator
+
+PLATFORMS: list[Platform] = [Platform.MEDIA_PLAYER]
+
+
+type PlaystationNetworkConfigEntry = ConfigEntry[PlaystationNetworkCoordinator]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: PlaystationNetworkConfigEntry
+) -> bool:
+    """Set up Playstation Network from a config entry."""
+
+    try:
+        psn = PSNAWP(entry.data["npsso"])
+    except PSNAWPAuthenticationError as error:
+        raise ConfigEntryAuthFailed(error) from error
+    except Exception as ex:
+        raise ConfigEntryNotReady(ex) from ex
+
+    try:
+        user = await hass.async_add_executor_job(lambda: psn.user(online_id="me"))
+    except PSNAWPAuthenticationError as error:
+        raise ConfigEntryAuthFailed(error) from error
```

Comment:
```suggestion
        raise ConfigEntryNotReady(error) from error
```

See review comment from #1